modules/core/include/opencv2/core/base.hpp

```diff
@@ -57,6 +57,77 @@
 #include "opencv2/core/cvdef.h"
 #include "opencv2/core/cvstd.hpp"
 
+#ifdef CV_COMPATIBLE_API
+
+namespace std
+{
+    CV_DEPRECATED
+    static inline const int& max(const ElemDepth& left, const int& right)
+    {
+        return max(static_cast<int>(left), right);
```

Comment:
Return type is invalid here.
Never return reference on local objects.

Also "const reference" in arguments is useless and problematic for enum/int/other scalar types.

P.S. For better understanding, try to interpret reference as a syntax sugar for "pointers".